### PR TITLE
Notice for renaming `none()` to `discard()`

### PR DESCRIPTION
**Pull request by ryn5**
Provides notice in release notes that none() will be renamed to discard() in https://github.com/apache/tinkerpop/pull/2377.

### CODE REVIEW

**Comment by Cole-Greer** [comment]
LGTM VOTE +1

File: docs/src/upgrade/release-3.7.x.asciidoc [hunk 1]
```diff
@@ -30,9 +30,15 @@ complete list of all the modifications that are part of this release.
 
 === Upgrading for Users
 
+==== Renaming `none()` Step to `discard()`
```
Comment (xiazcy) [inline docs/src/upgrade/release-3.7.x.asciidoc:33]:
```suggestion
==== Notice: Renaming `none()` step to `discard()` in 4.0.0
```

File: docs/src/upgrade/release-3.7.x.asciidoc [hunk 2]
```diff
@@ -30,9 +30,15 @@ complete list of all the modifications that are part of this release.
...
+`none()`, which is primarily used by `iterate()` to discard traversal results in remote contexts, will be renamed to
+`discard()` in release 4.0.0.
 
 === Upgrading for Providers
 
+==== Renaming NoneStep to DiscardStep
```
Comment (xiazcy) [inline docs/src/upgrade/release-3.7.x.asciidoc:39]:
```suggestion
==== Notice: Renaming NoneStep to DiscardStep in 4.0.0
```

**Review by xiazcy — COMMENTED** [review]
VOTE +1.
Just minor nit to make it clearer in the title.

**Comment by vkagamlyk** [comment]
VOTE+1

**Pull request merged**